library/ns1_record.py

```diff
@@ -548,8 +549,10 @@ def record_exit(self, after=None, before=None, changed=None, record=None):
         :type : Bool
         """
         # convert dictionaries to yaml txt dump
-        before_yaml = yaml.safe_dump(before, default_flow_style=False)
-        after_yaml = yaml.safe_dump(after, default_flow_style=False)
+        yaml=YAML(typ='safe')
+        yaml.default_flow_style = False
+        before_yaml = yaml.dump(before, sys.stdout)
+        after_yaml = yaml.dump(after, sys.stdout)
```

Comment:
Streaming to stdout breaks the diff which can be seen when running the test.
For example here is the diff on the output of the task `Test answer append`
```json
"changed": true,
"diff": {
    "after": null,
    "before": null
},
```

Could you please change it to remove the `sys` import and use `io.StringIO`:

```python
from io import StringIO

with StringIO() as before_stream:
    yaml.dump(before, stream=before_stream)
    before_yaml = before_stream.getvalue()

with StringIO() as after_stream:
    yaml.dump(after, stream=after_stream)
    after_yaml = after_stream.getvalue()
```

Then we can see the correct diff in the test:
```json
"changed": true,
"diff": {
    "after": "answers:\n- answer:\n  - 192.168.1.0\n  meta:\n    up: true\n- answer:\n  - 192.168.1.1\n  meta:\n    up: true\ncreated_at: 1721384831\ndomain: www.ansible-paulburrows.com\nfeeds: []\nfilters:\n- config: {}\n  filter: up\nmeta: {}\nnetworks:\n- 0\nregions: {}\ntier: 2\nttl: 3600\ntype: A\nupdated_at: 1721384831\nuse_client_subnet: true\nzone: ansible-paulburrows.com\nzone_name: ansible-paulburrows.com\n",
    "before": "answers:\n- answer:\n  - 192.168.1.0\n  meta:\n    up: true\ncreated_at: 1721384831\ndomain: www.ansible-paulburrows.com\nfeeds: []\nfilters:\n- config: {}\n  filter: up\nmeta: {}\nnetworks:\n- 0\nregions: {}\ntier: 2\nttl: 3600\ntype: A\nupdated_at: 1721384831\nuse_client_subnet: true\nzone: ansible-paulburrows.com\nzone_name: ansible-paulburrows.com\n"
    },
```

Reply:
I've committed your suggestion, but have had difficulty getting a working test environment. Can you confirm it is functioning as expected now?

Reply:
Looks good, thank you!! 🚀